azure-quantum/azure/quantum/optimization/term.py

```diff
@@ -4,23 +4,45 @@
 ##
 
 from typing import List, Union, Optional
+import numpy as np
+import numpy.typing as npt
 
 __all__ = ['Term']
 
+numpy_integer_types = [np.byte, np.ubyte, np.short, np.ushort, np.intc, np.uintc,
+                       np.int_, np.uint, np.longlong, np.ulonglong, np.int8, np.uint8,
+                       np.int16, np.uint16, np.int32, np.uint32, np.int64, np.uint64]
+
+numpy_float_types = [np.float16, np.float32, np.float64, np.float_, np.half, np.single, np.double]
+
+def _convert_if_numpy_type(param : npt.DTypeLike):
```

Comment:
I suspect you want ArrayLike and not DTypeLike here? DTypeLike matches objects representing dtypes, and not values of those dtypes.

Reply:
Thanks!